internal/services/springcloud/spring_cloud_service_resource.go

```diff
@@ -351,11 +351,18 @@ func resourceSpringCloudServiceCreate(d *pluginsdk.ResourceData, meta interface{
 	}
 	d.SetId(id.ID())
 
-	log.Printf("[DEBUG] Updating Config Server Settings for %s..", id)
-	if err := updateConfigServerSettings(ctx, configServersClient, id, gitProperty); err != nil {
-		return err
+	skuName := d.Get("sku_name").(string)
+	if skuName == "E0" && gitProperty != nil {
```

Comment:
should this not be 
```suggestion
	if skuName == "E0" {
```
otherwise if gitProperty is set we'll still try and set it?

Reply:
gitProperty can't be set when skuName="E0", so I returned an error here. And when skuName is not "E0", user can set and remove the gitProperty.